cpp/tests/test_abm_testing_strategy.cpp

```diff
@@ -97,14 +97,15 @@ TEST(TestTestingScheme, runScheme)
 
     ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>> mock_uniform_dist;
     EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
-        .Times(testing::Exactly(4))
-        .WillOnce(testing::Return(0.7))
-        .WillOnce(testing::Return(0.5))
-        .WillOnce(testing::Return(0.7))
-        .WillOnce(testing::Return(0.5));
-    ASSERT_EQ(testing_scheme1.run_scheme(rng_person1, person1, start_date), false); // Person tests and tests positive
-    ASSERT_EQ(testing_scheme2.run_scheme(rng_person2, person2, start_date), true); // Person tests and tests negative
-    ASSERT_EQ(testing_scheme1.run_scheme(rng_person1, person1, start_date),
+        .Times(testing::Exactly(5))
+        .WillOnce(testing::Return(0.7)) // Person got tested
+        .WillOnce(testing::Return(0.7)) // Test is positive
+        .WillOnce(testing::Return(0.5)) // Person complies to isolation
+        .WillOnce(testing::Return(0.7)) // Person got tested
+        .WillOnce(testing::Return(0.5)); // Test is negative
+    EXPECT_EQ(testing_scheme1.run_scheme(rng_person1, person1, start_date), false); // Person tests and tests positive
+    EXPECT_EQ(testing_scheme2.run_scheme(rng_person2, person2, start_date), true); // Person tests and tests negative
+    EXPECT_EQ(testing_scheme1.run_scheme(rng_person1, person1, start_date),
               true); // Person doesn't test
```

Comment:
Yes, one thing I should note is that the person is trying to migrate to the same work location. I would expect the agent to try different location types if it's not able to enter a location. 
One solution is to have the agent store the previous test result. It will be done in #466.

Reply:
The issue here is different, essentially, having a compliance for quarantine here breaks testing, since people may refuse to quarantine after a positive test. However, *any* test taken within the test delay after the first, independent of compliance or result, will result in a "negative" test (i.e. `true`).

Since this issue is caused by changes made in this branch, it must be fixed before merge. If you want to make this fix in another branch, we will have to merge that branch first.

Reply:
Yes, I think I understand now. I think it is fixed with the stored TestResult in https://github.com/SciCompMod/memilio/issues/466, where minimal_time_since_last_test will be replaced by a test's validity period and the agent will have to refer to the past result within this time. 
I will merge with it when the review is finished.